Frameworks/UIKit/UIApplication.mm

```diff
@@ -778,10 +778,22 @@ - (void)sendEvent:(UIEvent*)event {
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes Windows Launcher only supports launching URIs asynchronously. Check if the URI can be opened
+ and return its result to the caller instead of blocking for the URI to be actually opened.
 */
 - (BOOL)openURL:(NSURL*)url {
-    return [_launcher _openURL:url];
+    if ([self canOpenURL:url]) {
+        dispatch_queue_t queue =
+            ([NSThread isMainThread]) ? dispatch_get_main_queue() : dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0);
+        dispatch_async(queue,
+                       ^{
+                           [_launcher _openURL:url];
```

Comment:
Activation can be called from a non-UI thread and the launcher's _openURL can block in that case. We have not evaluated all the URI activation scenarios for PLM yet and I did not want to change the launcher until we do that; hence this local change.

---
In reply to: [72368518](https://github.com/Microsoft/WinObjC/pull/692#discussion_r72368518) [](ancestors = 72368518)